internal/core/port/user_port.go

```diff
@@ -0,0 +1,19 @@
+package port
+
```

Comment:
I think it is more organized to eliminate this package and move the interfaces to where it is consumed.